File: src/main/java/takumi3s/rcdamagecore/command/TestCommand.java
package takumi3s.rcdamagecore.command;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;
import takumi3s.rcdamagecore.util.LevelUtil;

public class TestCommand implements CommandExecutor {
    @Override
    public boolean onCommand(@NotNull CommandSender commandSender, @NotNull Command command, @NotNull String s, @NotNull String[] strings) {
//        var gson = new GsonBuilder().serializeNulls().disableHtmlEscaping().create();
//        commandSender.sendMessage(Component.text(gson.toJson(MythicBukkit.inst().getItemManager().getItem(strings[0]).get())));
        if(commandSender instanceof Player player) {
            player.getEquipment().getItemInMainHand().editMeta(meta -> {
//                LevelUtil.IS_LVL_ITEM.set(meta.getPersistentDataContainer(), true);
                LevelUtil.LVL.set(meta.getPersistentDataContainer(), 10);
                LevelUtil.MAX_LVL.set(meta.getPersistentDataContainer(), 20);
//                meta.getPersistentDataContainer().set(LevelUtil.IS_LEVELING_ITEM, PersistentDataType.BOOLEAN, true);
//                meta.getPersistentDataContainer().set(LevelUtil.LEVEL, PersistentDataType.INTEGER, 10);
//                meta.getPersistentDataContainer().set(LevelUtil.MAX_LEVEL, PersistentDataType.INTEGER, 20);
            });
            commandSender.sendMessage(Component.text("Success.").color(NamedTextColor.GREEN));
        } else {
            commandSender.sendMessage("Failure.");
        }
        return true;
    }
}

File: src/main/kotlin/net/azisaba/rcdamagecore/command/TestCommand.kt
package net.azisaba.rcdamagecore.command

import co.aikar.commands.BaseCommand
import co.aikar.commands.annotation.CommandAlias
import co.aikar.commands.annotation.CommandPermission
import co.aikar.commands.annotation.Subcommand
import net.azisaba.rcdamagecore.data.ItemData
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import org.bukkit.entity.Player

@CommandAlias("rdctest")
class TestCommand : BaseCommand() {
    @Subcommand("adddata")
    @CommandPermission("rcdamagecore.cmd.test.adddata")
    fun addDataForLevelingItem(player: Player) {
        player.equipment.itemInMainHand.editMeta {
            // apply data {exp: 0, max_level: 100} to held item
            ItemData.EXP.set(it.persistentDataContainer, 0)
            ItemData.MAX_LEVEL.set(it.persistentDataContainer, 100)
        }
        player.sendMessage(Component.text("Success to apply data.").color(NamedTextColor.GREEN))
    }
}
